lib/rules/no-duplicate-selectors/__tests__/index.js

```diff
@@ -84,6 +84,12 @@ testRule(rule, {
     message: messages.rejected("a b"),
     line: 1,
     column: 12,
+  }, {
+    code: "a { & {} }",
+    description: "duplicate caused by &-parent selector",
+    message: messages.rejected("a"),
```

Comment:
Worth noting that [CSS Nesting Module Level 3](https://tabatkins.github.io/specs/css-nesting/) it's just a Tab Atkin's idea. It's not related to W3C. Basically it have the same “power” as any blog post.

I think postcss-resolve-nested-selector is working well and resolves `&` well.